modules/drivers/sensors/ltr553als/ltr553als.js

```diff
@@ -0,0 +1,342 @@
+      Register.PS_CONTR,
+      (value & ~PS_MODE_MASK) | ((mode << PS_MODE_SHIFT) & PS_MODE_MASK)
+    );
+  }
+
+  #setAlsMode(mode) {
+    const io = this.#io;
+    let value = io.readUint8(Register.ALS_CONTR);
+    io.writeUint8(
+      Register.ALS_CONTR,
+      (value & ~ALS_MODE_MASK) | ((mode << ALS_MODE_SHIFT) & ALS_MODE_MASK)
+    );
+  }
+
+  configure(options) {
+    const io = this.#io;
+
+    this.#setAlsMode(AlsMode.ALS_STAND_BY_MODE);
+    this.#setPsMode(PsMode.PS_STAND_BY_MODE);
+
+    if (undefined != options.psLedPulseFreq) {
+      let value = io.readUint8(Register.PS_LED);
+      io.writeUint8(
+        Register.PS_LED,
+        (value & ~LED_PULSE_FREQ_MASK) |
+          ((options.psLedPulseFreq << LED_PULSE_FREQ_SHIFT) &
+            LED_PULSE_FREQ_MASK)
+      );
+    }
+
+    if (undefined != options.psLedDutyCycle) {
+      let value = io.readUint8(Register.PS_LED);
+      io.writeUint8(
+        Register.PS_LED,
+        (value & ~LED_DUTY_CYCLE_MASK) |
+          ((options.psLedDutyCycle << LED_DUTY_CYCLE_SHIFT) &
+            LED_DUTY_CYCLE_MASK)
+      );
+    }
+
+    if (undefined != options.psMeasRate) {
+      io.writeUint8(
+        Register.PS_MEAS_RATE,
+        options.psMeasRate & PS_MEAS_RATE_MASK
+      );
+    }
+
+    if (undefined != options.psLedPeakCurrent) {
+      let value = io.readUint8(Register.PS_LED);
+      io.writeUint8(
+        Register.PS_LED,
+        (value & ~LED_PEAK_CURRENT_MASK) |
+          (options.psLedPeakCurrent & LED_PEAK_CURRENT_MASK)
+      );
+    }
+
+    if (undefined != options.psNPulses) {
+      io.writeUint8(Register.PS_N_PULSES, options.psNPulses & PS_PULSES_MASK);
+    }
+
+    if (undefined != options.alsGain) {
+      let value = io.readUint8(Register.ALS_CONTR);
+      io.writeUint8(
+        Register.ALS_CONTR,
+        (value & ~ALS_GAIN_MASK) |
+          ((options.alsGain << ALS_GAIN_SHIFT) & ALS_GAIN_MASK)
+      );
+    }
+
+    if (undefined != options.alsIntegrationTime) {
+      let value = io.readUint8(Register.ALS_MEAS_RATE);
+      io.writeUint8(
+        Register.ALS_MEAS_RATE,
+        (value & ~ALS_INTEGRATION_TIME_MASK) |
+          ((options.alsIntegrationTime << ALS_INTEGRATION_TIME_SHIFT) &
+            ALS_INTEGRATION_TIME_MASK)
+      );
+    }
+
+    if (undefined != options.alsMeasureRate) {
+      let value = io.readUint8(Register.ALS_MEAS_RATE);
+      io.writeUint8(
+        Register.ALS_MEAS_RATE,
+        (value & ~ALS_MEASURE_RATE_MASK) |
+          ((options.alsMeasureRate << ALS_MEASURE_RATE_SHIFT) &
+            ALS_MEASURE_RATE_MASK)
+      );
+    }
+
+    if (undefined != options.psMeasRate) {
+      let value = io.readUint8(Register.ALS_MEAS_PATE_REG);
+      io.writeUint8(Register.ALS_MEAS_PATE_REG, value & PS_MEAS_RATE_MASK);
+    }
+
+    this.#setPsMode(PsMode.PS_ACTIVE_MODE);
+    this.#setAlsMode(AlsMode.ALS_ACTIVE_MODE);
+  }
+  close() {
+    this.#io.close();
+    this.#io = undefined;
+  }
+
+  get psValue() {
```

Comment:
These there accessors are part of the public API of the sensor. I think they are just helper functions. Since they are only called in one place, maybe it is better to inline them into `sample()`. If you do that, you can also reuse `buffer` for all three reads, which will be a little lighter.